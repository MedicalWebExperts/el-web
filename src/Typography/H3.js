import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

const propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.oneOf(['primary', 'secondary']),
};

const defaultProps = {
  color: 'primary',
};

const styles = theme => ({
  headline: {
    fontSize: theme.fonts.baseSize * 1.8,
    color: props => theme.colors[props.color],
  },
});

const H3 = ({ classes, text }) => (
  <h3 className={classes.headline}>{text}</h3>
);

H3.propTypes = propTypes;
H3.defaultProps = defaultProps;

export default injectSheet(styles)(H3);
