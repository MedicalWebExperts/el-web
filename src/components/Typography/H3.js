import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  text: PropTypes.string.isRequired,
};

const H3 = ({ text }) => (
  <h1>{text}</h1>
);

H3.propTypes = propTypes;

export default H3;
