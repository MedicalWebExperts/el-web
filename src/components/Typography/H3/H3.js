import React from 'react';
import PropTypes from 'prop-types';
import { withTheme } from '../../../theme';

const propTypes = {
  text: PropTypes.string.isRequired,
  secondary: PropTypes.bool,
  color: PropTypes.oneOf(['primary', 'secondary', 'default']),
  theme: PropTypes.shape({}).isRequired,
  styles: PropTypes.shape({}),
};

const defaultProps = {
  color: 'default',
  secondary: false,
  styles: {},
};

const H3 = ({
  text,
  theme,
  color,
  secondary,
  styles,
}) => (
  <h3 style={styles}>
    <style jsx>{`
      h3 {
        color: ${theme.colors[color]}
        font-family: ${secondary ? theme.font.secondary : theme.font.primary}
        font-size: ${theme.font.size * theme.typography.h3.fontSize}px
        line-height: ${theme.font.size / 10}
        margin-top: 0
        margin-bottom: ${theme.spacing * theme.typography.h3.marginBottom}px
      }
    `}
    </style>
    {text}
  </h3>
);

H3.propTypes = propTypes;
H3.defaultProps = defaultProps;

export default withTheme(H3);
