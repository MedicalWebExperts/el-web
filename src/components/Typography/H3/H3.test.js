import React from 'react';
import renderer from 'react-test-renderer';

import H3 from './H3';

describe('Component H3 Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer.create(<H3 text="Heading 3" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('primary color of the theme', () => {
    const tree = renderer.create(<H3 text="Heading 3" color="primary" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('secondary font-family of the theme', () => {
    const tree = renderer.create(<H3 text="Heading 3" secondary />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('added optional style inline prop', () => {
    const myStyles = {
      marginTop: 30,
      lineHeight: 2,
    };
    const tree = renderer.create(<H3 text="Heading 1" styles={myStyles} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
