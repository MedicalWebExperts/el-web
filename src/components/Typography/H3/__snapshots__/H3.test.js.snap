// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component H3 Snapshot added optional style inline prop 1`] = `
<h3
  className="jsx-3975818565"
  style={
    Object {
      "lineHeight": 2,
      "marginTop": 30,
    }
  }
>
  Heading 1
</h3>
`;

exports[`Component H3 Snapshot primary color of the theme 1`] = `
<h3
  className="jsx-3289541199"
  style={Object {}}
>
  Heading 3
</h3>
`;

exports[`Component H3 Snapshot secondary font-family of the theme 1`] = `
<h3
  className="jsx-2469268504"
  style={Object {}}
>
  Heading 3
</h3>
`;

exports[`Component H3 Snapshot without arguments, take the default values 1`] = `
<h3
  className="jsx-3975818565"
  style={Object {}}
>
  Heading 3
</h3>
`;
